presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -167,18 +173,25 @@ public Page getOutput()
                 }
                 currentProbePositionProducedRow = false;
                 if (!advanceProbePosition()) {
+                    probeEnd = true;
                     break;
                 }
                 statisticsCounter.recordProbe(joinSourcePositions);
                 joinSourcePositions = 0;
             }
         }
 
-        // only flush full pages unless we are done
-        if (pageBuilder.isFull() || (finishing && !pageBuilder.isEmpty() && probe == null)) {
-            Page page = pageBuilder.build();
+        // flush a page if we are done, or when probe reached the end;
+        // dictionary block's indices cannot span multiple probe pages.
+        // No need to check (finishing && !pageBuilder.isEmpty()) && probeEnd) given it implies probeEnd
```

Comment:
Not sure this "No need to check ..." comment is needed or helpful. It's clear to someone who knows previous code version of who reads the diff, but not later. Maybe 

* just drop it OR
* be higher-level, eg "No need to special-handle `finishing` state, checking `probeEnd` is sufficient to correctly produce last output page"